### ISSUE TITLE
Error when using recursive populate on mongoose document element with empty array 

### ISSUE THREAD

**Issue by Creiger** [feathersjs/feathers#583]
### Steps to reproduce
Create 3 mongoose documents with references ex. channels,users,groups. Channel can have 0:n users, user can have 0:1 groups. Channels schema has users defined like this:
`users: [{type: mongooseClient.Schema.Types.ObjectId, ref: 'users', required: true}]`
so when you create new Channel document, without any users, mongoose adds by design
`users: []`
to new document. Then add populate schema to channel hooks
`include: {
        service: 'users',
        nameAs: 'users',
        parentField: 'users',
        childField: '_id',
        include: [
            {
                service: 'groups',
                nameAs: 'group',
                parentField: 'group',
                childField: '_id'
            }
        ]
    }`
and then fetch channels.

You will get an error:

> Cannot read property 'toJSON' of null
error:  TypeError: Cannot read property 'toJSON' of null
    at populateItemArray (/data/www/lukashronec.cz/ptt/node_modules/feathers-hooks-common/lib/services/populate.js:108:12)
    at /data/www/lukashronec.cz/ptt/node_modules/feathers-hooks-common/lib/services/populate.js:234:14
    at process._tickCallback (internal/process/next_tick.js:103:7)

I also tried non recursive populate with users only and it worked. When i remove those Channels users where they are empty, it works with recursive schema. So only recursive population has problem with empty arrays.

### Expected behavior
I would expect that populateItemArray would check if items are an empty array and return again an empty array. Same when i use non recursive populate schema.

### Actual behavior
It throws an error.

### System configuration

Tell us about the applicable parts of your setup.

> "feathers": "^2.1.2",
    "feathers-authentication": "^1.2.3",
    "feathers-authentication-hooks": "^0.1.3",
    "feathers-authentication-jwt": "^0.3.1",
    "feathers-authentication-local": "^0.3.4",
    "feathers-configuration": "^0.4.1",
    "feathers-errors": "^2.8.0",
    "feathers-hooks": "^2.0.1",
    "feathers-hooks-common": "^3.3.2",
    "feathers-mongoose": "^5.1.0",
    "feathers-rest": "^1.7.2",
    "feathers-socketio": "^2.0.0",

**Comment by eddyystop** [comment]
You will find your issues might be handled faster is they are posted to the repo they affect, feathers-hooks-common in this case.

Preliminaries for fix are in https://github.com/feathersjs/feathers-hooks-common/pull/190

**Comment by eddyystop** [comment]
I think you would want to use `asArray` in this case to ensure you always get an array back, even if it has just 1 element. It looks like that would avoid the problem until a fix lands.

```javascript
include: {
  service: 'users',
  nameAs: 'users',
  parentField: 'users',
  childField: '_id',
  asArray: true, // workaround
  include: [ ... ]
}
```

**Comment by eddyystop** [comment]
Fixed in https://github.com/feathersjs/feathers-hooks-common/pull/191 which will be published as v3.3.3.

Please confirm the fix works for you. Could you afterwards close this issue? Thanks.

**Comment by Creiger** [comment]
Thank you eddyystop and sorry for the wrong repo.
